internal/brokers/broker.go

```diff
@@ -369,7 +369,11 @@ func validateUserInfoAndGenerateIDs(selectedUsername string, uInfo userInfo) (us
 	if uInfo.Name == "" {
 		return users.UserInfo{}, fmt.Errorf("empty username")
 	}
-	if uInfo.Name != selectedUsername {
+	// Ensure that the name of the user which authenticated matches
+	// the name which the user tries to log in as.
+	// Microsoft Entra user names are case insensitive, so we ignore
```

Comment:
I wouldn't mention Microsoft Entra ID in the comment. This is `authd` codebase, so it should talk about brokers in a more "generic" way. Maybe go for something like "some brokers can be case insensitive, so we ignore the case when comparing the names"?

Reply:
I tried to make it a bit more generic, but I think it's important that we don't ignore the case in the user name if we support a broker which doesn't ignore the case (for the same reason explained in https://github.com/ubuntu/authd/pull/511#discussion_r1741235776, i.e. to avoid collisions which allow impersonation), so I made it less vague about the supported brokers having case insensitive names.

Reply:
that's obsolete now